### DISCUSSION TITLE
FYI all: New IDE version 2.3.0 is out. I was personally informed that this fixes a number of relevant bugs that were impacting us

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/ATTinyCore#841] [General]
Most notably, it should be possoble to install the core automatically on a 2.x.x version. I have yet to test this

**Comment by hmeijdam** [comment]
Installation of Attinycore and Megatinycore worked like a breeze on 2.3.0

Still no portable version of the V2 IDE, so it's in the recycle bin already.

**Reply by bxparks** [reply]
By "portable version", do you mean https://docs.arduino.cc/software/ide-v1/tutorials/PortableIDE/ ? Yeah, I depend on that too.

**Reply by hmeijdam** [reply]
Yes, I have currently 17 different versions of the IDE V1 working next to each other on my PC.  Sometimes I need to go back to a very old version, or have a version where the source is tweaked.

**Comment by ahshah** [comment]
Any idea of which bugs or issues it fixes?

**Reply by hmeijdam** [reply]
[Here](https://github.com/arduino/arduino-ide/releases/tag/2.3.0) are the releasenotes

**Reply by ahshah** [reply]
Thanks, however, the release notes don't really tell me how it pertains to the attiny codeline. 

Heres the bug fix list:

1. Forward backend logging to electron (https://github.com/arduino/arduino-ide/pull/2236)
2. Arduino state data not available to extensions after IDE startup (https://github.com/arduino/arduino-ide/pull/2233)
3. "Configure and Upload" dialog not presented for uploads dependent on "user provided fields" after IDE startup (https://github.com/arduino/arduino-ide/issues/2230)
4. Custom board option selection made in one window affects other windows as well (https://github.com/arduino/arduino-ide/issues/2240)
5. "Start Debugging" button incorrectly disabled if board was pre-selected on IDE startup (https://github.com/arduino/arduino-ide/issues/2237)
6. Board-dependent menus not populated if board was pre-selected on IDE startup (https://github.com/arduino/arduino-ide/issues/2239)
7. Saving sketch fails silently when files are read-only (https://github.com/arduino/arduino-ide/issues/1501)
8. IDE window doesn't open when started via sketch file association (https://github.com/arduino/arduino-ide/issues/2209)
9. Subfolder structure of sketch lost after "Save As" operation (https://github.com/arduino/arduino-ide/issues/2077)
10. Naming collision during "Save As" causes overwrite of secondary sketch file (https://github.com/arduino/arduino-ide/issues/827)
11. Previous selection remains in "Firmware Updater" and "Certificate Uploader" menu after no longer available (https://github.com/arduino/arduino-ide/issues/2222)
12. Bundle native Arduino Firmware Uploader with Apple Silicon build (https://github.com/arduino/arduino-ide/pull/2338)

**Reply by per1234** [reply]
Hi @ahshah. The release notes in the `arduino/arduino-ide` repository are specific to the changes in that repo's codebase. Those include bumps to the IDE's tool dependencies, but the changes in those tools that were shipped to IDE users through the dependency bumps are not mentioned there (you would need to check the release notes for the individual tools, most significantly [Arduino CLI](https://github.com/arduino/arduino-cli)) to get the full picture of what changes you would receive from updating the IDE.

Because the distinction between the IDE application and its bundled tool dependendencies is not something the end user is aware of or interested in, I make an attempt to list the changes from the work in the IDE codebase as well as the tools in the release announcement that is made on Arduino Forum:

https://forum.arduino.cc/t/arduino-ide-2-3-0-is-now-available/1221189
